### PR TITLE
[#2 Software Updates] Add Phases struct

### PR DESCRIPTION
**Pull request by MatApple**

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @rcorral

File: src/js/structs/__tests__/ServicePlan-test.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import ServicePlan from '../ServicePlan';
+import ServicePlanPhases from '../ServicePlanPhases';
```
Comment (jfurrow) [inline src/js/structs/__tests__/ServicePlan-test.js]: This is failing for me.

File: src/js/structs/ServicePlanPhases.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  constructor() {
+    super(...arguments);
+
+    // Cache lookups
+    this._activePhaseIndex = -1;
+    this._activePhase = null;
+
+    // Find first phase where status is not complete
+    this.getItems().some((phase, phaseIndex) => {
+      if (!(phase instanceof ServicePlanPhase)) {
```
Comment (kennyt) [inline src/js/structs/ServicePlanPhases.js]: we never completely transform the items in `this.list` into `ServicePlanPhase`s anywhere. we just create a phase to check if it's complete

File: src/js/structs/ServicePlanPhases.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    // Cache lookups
+    this._activePhaseIndex = -1;
+    this._activePhase = null;
+
+    this.list = this.getItems().map(function (phase) {
+      if (!(phase instanceof ServicePlanPhase)) {
+        return new ServicePlanPhase(phase);
+      }
+      return phase;
```
Comment (mlunoe) [inline src/js/structs/ServicePlanPhases.js]: #nit newline before this `return`

Comment (MatApple) [inline src/js/structs/ServicePlanPhases.js]: fixed

**Pull request merged**